sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraServiceImpl.java

```diff
@@ -320,4 +320,46 @@ public TokenRange(
       this.rangeEnd = rangeEnd;
     }
   }
+
+  /**
+   * Writer storing an entity into Apache Cassandra database.
+   */
+  protected class WriterImpl<T> implements Writer<T> {
+
+    private final CassandraIO.Write<T> spec;
+
+    private final Cluster cluster;
+    private final Session session;
+    private final MappingManager mappingManager;
+
+    public WriterImpl(CassandraIO.Write<T> spec) {
+      this.spec = spec;
+      this.cluster = getCluster(spec.hosts(), spec.port());
+      this.session = cluster.connect(spec.keyspace());
+      this.mappingManager = new MappingManager(session);
+    }
+
+    @Override
+    public void write(T entity) {
+      Mapper<T> mapper = (Mapper<T>) mappingManager.mapper(entity.getClass());
+      mapper.save(entity);
```

Comment:
This is synchronous, right? i.e. on one hand, after the call to save() completes, the entity is guaranteed to be reliably committed to Cassandra; on the other hand, that means it's kinda slow? (as opposed to saveAsync() I suppose)

This is fine - just please add a comment, if my understanding is correct.